### DISCUSSION TITLE
Save register user IP on database

### DISCUSSION THREAD

**Discussion by karaka200** [symfony/symfony#44694] [Q&A] [answered]
When a user register, I want add user IP on my database.
On User entity add these :

```
#[ORM\Column(type: 'string', length: 255)]
private ?string $regIP;

public function getRegIP(): ?string
{
    return $this->regIP;
}

public function setRegIP(): self
{
    $request = Request::createFromGlobals();
    $this->regIP = $request->getClientIp();
    return $this;
}
```

Are these codes appropriate for this topic?
When I try to add a user with `symfony console doctrine:fixtures:load`, it gives an error and returns NULL

**Comment by stof** [comment, marked as answer]
Well, in the console, you cannot create a request from globals, as you are not in the context of an HTTP request.

Thus, you should not create a request from globals at all, but use the Request object being handled by the kernel (and the controller, and so on). This will make your code compatible with code that does not rely on the superglobals for the request (for instance, tests).

**Reply by karaka200** [reply]
So do you think I should use this code or do you have something else?

```
private Request $request;
public function __construct(Request $request){
    $this->request = $request;
}

public function setRegIP(): self
{
    $this->regIP = $this->request->getClientIp();
    return $this;
}
```

**Reply by alessandro-podo** [reply]
use a setter only and set the IP in your register Service

**Reply by stof** [reply]
don't pass the request in the constructor of your user (that's not where it belongs, and it will cause issues when loading the user from the DB). Pass it to the method instead.

**Reply by stof** [reply]
or even better, pass only the IP to the setter, allowing you to set a fake IP in your fixtures without having to build a fake request for that.
